src/hub/Hub.sol

```diff
@@ -95,32 +92,23 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler {
         }
     }
 
-    /// @inheritdoc IHub
-    function execute(PoolId poolId, bytes[] calldata data) external payable {
-        require(unlockedPoolId.isNull(), IHub.PoolAlreadyUnlocked());
-        require(hubRegistry.isAdmin(poolId, msg.sender), IHub.NotAuthorizedAdmin());
-
-        accounting.unlock(poolId);
-        unlockedPoolId = poolId;
-
-        multicall(data);
-
-        accounting.lock();
-        unlockedPoolId = PoolId.wrap(0);
-    }
-
     //----------------------------------------------------------------------------------------------
     // Permissionless methods
     //----------------------------------------------------------------------------------------------
 
     /// @inheritdoc IHub
-    function createPool(address admin, AssetId currency) external payable auth returns (PoolId poolId) {
+    function createPool(address admin, AssetId currency) external payable returns (PoolId poolId) {
+        _auth();
+
         poolId = hubRegistry.registerPool(admin, sender.localCentrifugeId(), currency);
     }
 
     /// @inheritdoc IHub
-    function claimDeposit(PoolId poolId, ShareClassId scId, AssetId assetId, bytes32 investor) external payable {
-        _protected();
+    function claimDeposit(PoolId poolId, ShareClassId scId, AssetId assetId, bytes32 investor)
+        external
+        payable
+        protected
```

Comment:
Why protected?

Reply:
I think all of those can be public.

Reply:
`protected` is just the re-entrancy modifier

Reply:
Maybe we should rename it to reentrancy, I also found this name quite confusing. WDYT @hieronx ?